### ISSUE TITLE
Field.getAnnotation should be @Nullable

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#197]
```
What steps will reproduce the problem?
1.
Field field = ...;
SomeAnnotation a = field.getAnnotation(SomeAnnotation.class);
if(a != null)
...

What is the expected output? What do you see instead?
One gets a warning that the check is not needed, but like the API states null 
is a possible return value.
The framework shouldn't throw a warning


What version of the product are you using?
1.4.3

```

Original issue reported on code.google.com by `DannyNul...@gmail.com` on 20 Jan 2013 at 5:19
